extensions/wikia/WallNotifications/WallNotificationEntity.class.php

```diff
@@ -78,174 +127,202 @@ public function getUniqueId() {
 		$acParent = $wm->getTopParentObj();
-		$this->data->parent_username = '';
-		$this->data->thread_title = '';
-		$this->data_noncached->parent_title_dbkey = '';
 
-		$this->data_noncached->msg_text = $wm->getText();
-		$this->data->notifyeveryone = $wm->getNotifyeveryone();
+		if ( empty( $acParent ) ) {
+			$this->threadTitleFull = $wm->getMetaTitle();
 
-		if ( $wm->isEdited() ) {
-			$this->data->reason = $wm->getLastEditSummery();
+			$data->parent_id = null;
+			$data->thread_title = $wm->getMetaTitle();
+			$data->parent_username = $data->wall_username;
 		} else {
-			$this->data->reason = '';
+			$acParent->load();
+			$title = $acParent->getTitle();
+			$this->parentTitleDbKey = $title->getDBkey();
+			$this->threadTitleFull = $acParent->getMetaTitle();
+
+			$this->setMessageParentUserData( $data, $acParent );
+			$data->parent_id = $acParent->getId();
+			$data->thread_title = $acParent->getMetaTitle();
 		}
+	}
 
-		if ( !empty( $acParent ) ) {
-			$acParent->load();
-			$parentUser = $acParent->getUser();
-
-			if ( $parentUser instanceof User ) {
-				$this->data->parent_username = $parentUser->getName();
-				if ( $parentUser->getId() > 0 ) {
-					$this->data->parent_displayname = $this->data->parent_username;
-				} else {
-					$this->data->parent_displayname = wfMessage( 'oasis-anon-user' )->text();
-				}
-				$this->data->parent_user_id = $acParent->getUser()->getId();
+	private function setMessageParentUserData( stdClass $data, WallMessage $parent ) {
+		$parentUser = $parent->getUser();
+
+		if ( $parentUser instanceof User ) {
+			$data->parent_username = $parentUser->getName();
+			$data->parent_user_id = $parentUser->getId();
+
+			if ( $data->parent_user_id > 0 ) {
+				$data->parent_displayname = $data->parent_username;
 			} else {
+				$data->parent_displayname = wfMessage( 'oasis-anon-user' )->text();
+			}
+		} else {
 			/* parent was deleted and somehow reply stays in the system
 			 * the only way I've reproduced it was: I deleted a thread
 			 * then I went to Special:Log/delete and restored only its reply
 			 * an edge case but it needs to be handled
 			 * --nAndy
 			 */
-				$this->data->parent_username = $this->data->parent_displayname = wfMessage( 'oasis-anon-user' )->text();
-				$this->data->parent_user_id = 0;
-			}
-			$title = $acParent->getTitle();
-			$this->data_noncached->thread_title_full = $acParent->getMetaTitle();
-			$this->data->thread_title = $acParent->getMetaTitle();
-			$this->data_noncached->parent_title_dbkey = $title->getDBkey();
-			$this->data->parent_id = $acParent->getId();
-			$this->data->url = $wm->getMessagePageUrl();
-
-		} else {
-			$this->data->url = $wm->getMessagePageUrl();
-			$this->data->parent_username = $walluser->getName();
-			$this->data_noncached->thread_title_full = $wm->getMetaTitle();
-			$this->data->thread_title = $wm->getMetaTitle();
-		}
-
-		return true;
-	}
-
-	protected function buildId( $wikiId, $RCid ) {
-
-	}
-
```

Comment:
I thought ```recreateFromDB``` was a weird name and totally didn't match the other ```createFromXXXX``` methods that do similar things.  So I renamed this ```createFromId``` (since that's what it does given an entity ID) and moved it up next to the other create methods.